server/util/networking/networking.go

```diff
@@ -443,23 +429,56 @@ func SetupVethPair(ctx context.Context, netNamespace string) (_ *VethPair, err e
 		}
 	}
 
-	err = runCommand(ctx, "iptables", "--wait", "-A", "FORWARD", "-i", veth1, "-o", device, "-j", "ACCEPT")
-	if err != nil {
-		return nil, err
+	iptablesRules := [][]string{
+		// Allow forwarding traffic from the host side of the veth pair to the
+		// device associated with the configured route prefix (usually the
+		// default route). This is necessary on hosts with default-deny policies
+		// in place.
+		{"FORWARD", "-i", veth1, "-o", device, "-j", "ACCEPT"},
+
+		// Drop any traffic from the namespace that is targeting another
+		// namespace.
+		{"FORWARD", "-s", network.CIDR(), "-d", containerNetworkingCIDR, "-j", "DROP"},
```

Comment:
Containers shouldn't be addressing any private addresses so you might as well block all the ranges in `PrivateIPRanges`.

Reply:
We also have `executor.blackhole_private_ranges` which should disallow traffic to private ranges, but it felt somewhat incidental that `containerNetworkingCIDR` happens to be one of the private IP ranges so I thought it was worth an explicit rule. 

If we want to consolidate the handling of these DROP rules and maybe get rid of that blackhole_private_ranges flag, I can do that in a separate PR - wdyt?

Reply:
(Will discuss more offline and follow up in another PR)